lib/AST/ASTPrinter.cpp

```diff
@@ -109,6 +109,14 @@ PrintOptions PrintOptions::printTextualInterfaceFile() {
         }
       }
 
+      // Skip extensions that extend things we wouldn't print.
+      if (auto *ED = dyn_cast<ExtensionDecl>(D)) {
+        if (!shouldPrint(ED->getExtendedNominal(), options))
+          return false;
+        // FIXME: We also need to check the generic signature for constraints
```

Comment:
This is essentially already what the code in AccessRequests.cpp is doing. Isn't checking the extension's max access sufficient here?

Also I notice in shouldPrint() you have some special logic for typealiases. Please refactor this to use isPassthroughTypeAlias(), which is in Sema right now but could move to a method on TypeAliasDecl.

Reply:
Max access doesn't take UFI into account.

The logic that's there isn't for passthrough typealiases; it's for the typealiases synthesized in the body of generic types.

Reply:
I wonder if it would be better if we changed associated type inference to not create those special typealiases for generic parameters at all.

Reply:
Yep, when it first went in Doug mentioned that that's where we want to go. We can remove this then.